lib/srfi-152.scm

```diff
@@ -74,6 +74,9 @@
             (substring s cur end))))
 
 (define (string-segment s k)
+  (unless (and (exact-integer? k)
+               (positive? k))
+    (error "positive exact integer required, but got:" n))
```

Comment:
there is a typo (n instead of k)

Reply:
Ops, sorry!
I even tested, and got an error with the zero argument, but didn't pay close attention to what the error was ("unbounded var n", instead of the "positive k required")...
New version pushed